app/models/screening.rb

```diff
@@ -8,11 +8,9 @@ class Screening < ApplicationRecord
 
   def available_seats
     taken_seats = []
-    reservations.each do |r|
-      r.tickets.each do |t|
-        taken_seats.push(t.seat)
-      end
+    reservations.where.not(status: :cancelled).each do |reservation|
+      taken_seats << reservation.tickets.pluck(:seat)
```

Comment:
You could simplify this futher
```suggestion
   taken_seats = reservations.joins(:tickets).where.not(status: :cancelled).pluck(:'tickets.seat')
```